fix docker builds

- [ ] fix tests not running in docker
    + https://travis-ci.org/CCPPETMR/SIRF-SuperBuild/jobs/483407710#L5255
    + [x] ensure future failures will not be silently ignored
- [x] ~~fix docker login~~
    + https://travis-ci.org/CCPPETMR/SIRF-SuperBuild/jobs/483407710#L5234
    + ~~https://community.atlassian.com/t5/Bitbucket-questions/Piplenes-docker-login-can-not-perform-an-interactive-login-from/qaq-p/595736~~
    + https://discuss.bitrise.io/t/docker-login-failed-only-pr-trigger/2232
        * intentionally cannot build docker on PRs (security)

depends on #212 -> #213

OK I now have:

```
$ docker-compose -f docker-compose.yml -f docker-compose.nix.yml -f docker-compose.devel.yml build --build-arg UID=1000 --build-arg GROUPS=1000 sirf
Step 1/50 : FROM ubuntu:18.04 as base
...
Step 9/50 : FROM base as core
...
Step 31/50 : FROM core as sirf
...
Step 46/50 : RUN bash user_sirf-ubuntu.sh
$ git clone https://github.com/CCPPETMR/SIRF-SuperBuild --recursive -b master /opt/SIRF-SuperBuild
Cloning into '/opt/SIRF-SuperBuild'...
$ pushd /opt/SIRF-SuperBuild
/opt/SIRF-SuperBuild /home-away
$ echo $BUILD_FLAGS $EXTRA_BUILD_FLAGS
-DCMAKE_BUILD_TYPE=Release -DBUILD_STIR_WITH_OPENMP=ON -DUSE_SYSTEM_ACE=ON -DUSE_SYSTEM_Armadillo=ON -DUSE_SYSTEM_Boost=ON -DUSE_SYSTEM_FFTW3=ON -DUSE_SYSTEM_HDF5=ON -DUSE_ITK=ON -DBUILD_siemens_to_ismrmrd=ON -DUSE_SYSTEM_SWIG=ON -DDEVEL_BUILD=ON
$ cmake $BUILD_FLAGS $EXTRA_BUILD_FLAGS .
-- The C compiler identification is GNU 7.3.0
-- The CXX compiler identification is GNU 7.3.0
-- Check for working C compiler: /opt/pyvenv/bin/x86_64-conda_cos6-linux-gnu-cc
-- Check for working C compiler: /opt/pyvenv/bin/x86_64-conda_cos6-linux-gnu-cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /opt/pyvenv/bin/x86_64-conda_cos6-linux-gnu-c++
-- Check for working CXX compiler: /opt/pyvenv/bin/x86_64-conda_cos6-linux-gnu-c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- EXTERNAL_PROJECT_BUILD_TYPE: Release
-- Found PythonInterp: /opt/pyvenv/bin/python (found version "2.7.15")
-- Found PYTHON_EXECUTABLE=/opt/pyvenv/bin/python
-- Python version 2.7.15
-- Found PythonLibs: /opt/pyvenv/lib/libpython2.7.so (found version "2.7.15")
-- Found PYTHON_INCLUDE_DIRS=/opt/pyvenv/include/python2.7
-- Found PYTHON_LIBRARIES=/opt/pyvenv/lib/libpython2.7.so
-- Could NOT find Matlab (missing:  Matlab_INCLUDE_DIRS Matlab_MEX_LIBRARY Matlab_MEX_EXTENSION Matlab_ROOT_DIR MAIN_PROGRAM)
-- Python libraries found
-- SIRF and/or STIR Python modules will be installed in /opt/SIRF-SuperBuild/INSTALL/python
-- SuperBuild - First pass
-- Matlab_ROOT_DIR=
-- STIR_DIR=
-- NiftyReg_Binary_DIR=
-- SuperBuild - First pass - done
-- SuperBuild - CCPPETMR => Requires SIRF, STIR, Gadgetron, siemens_to_ismrmrd, NiftyReg,
-- Matlab_ROOT_DIR=
-- STIR_DIR=
-- NiftyReg_Binary_DIR=
-- SuperBuild -   SIRF => Requires STIR, Boost, HDF5, ISMRMRD, FFTW3, SWIG, NiftyReg,
-- SuperBuild -     STIR => Requires Boost, ITK,
-- SuperBuild -       Boost[OK]
-- Boost version: 1.65.1
-- USING the system Boost, using BOOST_LIBRARY_DIR=
-- SuperBuild -       ITK => Requires HDF5,
-- SuperBuild -         HDF5[OK]
-- HDF5: Using hdf5 compiler wrapper to determine C configuration
-- Found HDF5: /usr/lib/x86_64-linux-gnu/hdf5/serial/libhdf5.so;pthread;sz;z;dl;m (found suitable version "1.10.0.1", minimum required is "1.8")
-- USING the system HDF5, found HDF5_INCLUDE_DIRS=/usr/include/hdf5/serial, HDF5_C_LIBRARY_hdf5=/usr/lib/x86_64-linux-gnu/hdf5/serial/libhdf5.so,HDF5_LIBRARIES=/usr/lib/x86_64-linux-gnu/hdf5/serial/libhdf5.so;pthread;sz;z;dl;m
-- SuperBuild -       ITK[OK]
-- Adding project ITK
-- SuperBuild -     STIR[OK]
-- Adding project STIR
-- SuperBuild -     ISMRMRD => Requires HDF5[INCLUDED], Boost[INCLUDED], FFTW3,
-- SuperBuild -       FFTW3[OK]
FFTW3 UNIX libraries: FFTW3_LIBRARY-NOTFOUND
CMake Error at /opt/cmake/share/cmake-3.7/Modules/FindPackageHandleStandardArgs.cmake:138 (message):
  Could NOT find FFTW3 (missing: FFTW3_LIBRARY)
Call Stack (most recent call first):
  /opt/cmake/share/cmake-3.7/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  CMake/FindFFTW3.cmake:114 (find_package_handle_standard_args)
  SuperBuild/External_FFTW3.cmake:87 (find_package)
  CMake/ExternalProjectDependency.cmake:593 (include)
  SuperBuild/External_ISMRMRD.cmake:29 (ExternalProject_Include_Dependencies)
  CMake/ExternalProjectDependency.cmake:593 (include)
  SuperBuild/External_SIRF.cmake:37 (ExternalProject_Include_Dependencies)
  CMake/ExternalProjectDependency.cmake:593 (include)
  CMake/ExternalProjectDependency.cmake:639 (ExternalProject_Include_Dependencies)
  SuperBuild.cmake:210 (ExternalProject_Include_Dependencies)
  CMakeLists.txt:18 (include)

-- Configuring incomplete, errors occurred!
See also "/opt/SIRF-SuperBuild/CMakeFiles/CMakeOutput.log".
Makefile:45: recipe for target 'build' failed
```

Is this #147? I've already installed `libfftw3-dev`

```
_reg_maths_eigen.cpp:(.text._Z3svdIfEvPPT_mmPS2_S3_S3_[_Z3svdIfEvPPT_mmPS2_S3_S3_]+0x5e9): undefined reference to `GOMP_parallel'
/opt/SIRF-SuperBuild/INSTALL/lib/lib_reg_maths.a(_reg_maths_eigen.cpp.o):_reg_maths_eigen.cpp:(.text._Z3svdIfEvPPT_mmPS2_S3_S3_[_Z3svdIfEvPPT_mmPS2_S3_S3_]+0x610): more und
efined references to `GOMP_parallel' follow
collect2: error: ld returned 1 exit status
src/Registration/cReg/CMakeFiles/sirf_tensor_split_join.dir/build.make:126: recipe for target 'src/Registration/cReg/sirf_tensor_split_join' failed
make[5]: *** [src/Registration/cReg/sirf_tensor_split_join] Error 1
CMakeFiles/Makefile2:758: recipe for target 'src/Registration/cReg/CMakeFiles/sirf_tensor_split_join.dir/all' failed
make[4]: *** [src/Registration/cReg/CMakeFiles/sirf_tensor_split_join.dir/all] Error 2
make[4]: *** Waiting for unfinished jobs....
[ 67%] Building CXX object src/xSTIR/pSTIR/CMakeFiles/_pystir.dir/opt/SIRF-SuperBuild/sources/STIR/src/modelling_buildblock/modelling_registries.cxx.o
[ 68%] Building CXX object src/xSTIR/pSTIR/CMakeFiles/_pystir.dir/opt/SIRF-SuperBuild/sources/STIR/src/spatial_transformation_buildblock/spatial_transformation_registries.c
xx.o
[ 69%] Linking CXX shared module _pystir.so
[ 69%] Built target _pystir
make[3]: *** [all] Error 2
Makefile:138: recipe for target 'all' failed
CMakeFiles/SIRF.dir/build.make:118: recipe for target 'builds/SIRF/stamp/SIRF-build' failed
make[2]: *** [builds/SIRF/stamp/SIRF-build] Error 2
make[1]: *** [CMakeFiles/SIRF.dir/all] Error 2
CMakeFiles/Makefile2:306: recipe for target 'CMakeFiles/SIRF.dir/all' failed
Makefile:94: recipe for target 'all' failed
make: *** [all] Error 2
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -65,12 +65,12 @@ matrix:
    # +boost +itk -hdf5 +swig
    env: EXTRA_BUILD_FLAGS="-DUSE_SYSTEM_Boost=ON -DUSE_ITK=ON -DUSE_SYSTEM_HDF5=OFF -DUSE_SYSTEM_SWIG=ON" MATRIX_EVAL="CC=gcc CXX=g++" PYMVER=2
  # docker
- #- os: linux
- #  # +armadillo +boost +fftw3 +hdf5 +siemens_to_ismrmrd +swig
- #  env: DOCKER_BUILD=1
- #- os: linux
- #  # +DEVEL +armadillo +boost +fftw3 +hdf5 +siemens_to_ismrmrd +swig
- #  env: DOCKER_BUILD=DEVEL
+ - os: linux
+   # +ace +armadillo +boost +fftw3 +hdf5 +itk +siemens_to_ismrmrd +swig
+   env: DOCKER_BUILD=1
+ - os: linux
+   # +DEVEL +ace +armadillo +boost +fftw3 +hdf5 +itk +siemens_to_ismrmrd +swig
+   env: DOCKER_BUILD=DEVEL
 
 env:
  global:
@@ -231,7 +231,7 @@ before_install:
  # N.B.: don't put into build matrix to allow caching.
  - BUILD_FLAGS="$BUILD_FLAGS -DPYVER=$PYMVER"
  - cmake --version
- 
+
 install:
  - |
     if [[ -n "$DOCKER_BUILD" ]]; then
@@ -268,7 +268,7 @@ install:
         $PY_EXE -m pip install --user --only-binary=numpy,scipy,matplotlib numpy scipy matplotlib
         cmake $BUILD_FLAGS $EXTRA_BUILD_FLAGS .
         # Job may timeout (>50min) if no ccache, otherwise should be <1min:
-        make 
+        make
         mv INSTALL/share/gadgetron/config/gadgetron.xml.example INSTALL/share/gadgetron/config/gadgetron.xml
       ) || travis_terminate 1
       source $PWD/INSTALL/bin/env_ccppetmr.sh

--- a/.travis.yml
+++ b/.travis.yml
@@ -219,17 +219,11 @@ before_install:
  - $PY_EXE --version
  - $PY_EXE -m pip --version
  - $PY_EXE -m pip freeze
- # ccache compiler override
- - ln -s "$(which ccache)" g++
- - ln -s "$(which ccache)" g++-6
- - ln -s "$(which ccache)" g++-7
- - ln -s "$(which ccache)" gcc
- - ln -s "$(which ccache)" gcc-6
- - ln -s "$(which ccache)" gcc-7
  - export PATH="$PWD:$PATH"
  - popd
  # N.B.: don't put into build matrix to allow caching.
  - BUILD_FLAGS="$BUILD_FLAGS -DPYVER=$PYMVER"
+ - BUILD_FLAGS="$BUILD_FLAGS -DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache"
  - cmake --version
 
 install:

--- a/.travis.yml
+++ b/.travis.yml
@@ -219,11 +219,17 @@ before_install:
  - $PY_EXE --version
  - $PY_EXE -m pip --version
  - $PY_EXE -m pip freeze
+ # ccache compiler override
+ - ln -s "$(which ccache)" g++
+ - ln -s "$(which ccache)" g++-6
+ - ln -s "$(which ccache)" g++-7
+ - ln -s "$(which ccache)" gcc
+ - ln -s "$(which ccache)" gcc-6
+ - ln -s "$(which ccache)" gcc-7
  - export PATH="$PWD:$PATH"
  - popd
  # N.B.: don't put into build matrix to allow caching.
  - BUILD_FLAGS="$BUILD_FLAGS -DPYVER=$PYMVER"
- - BUILD_FLAGS="$BUILD_FLAGS -DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache"
  - cmake --version
 
 install:

--- a/.travis.yml
+++ b/.travis.yml
@@ -184,7 +184,7 @@ before_install:
         brew install curl
         #brew install cmake # already present
         # alternative: get our own
-        #curl -L -O https://cmake.org/files/v3.8/cmake-3.8.0-Darwin-x86_64.tar.gz
+        #curl -L -O https://github.com/Kitware/CMake/releases/download/v3.13.4/cmake-3.13.4-Darwin-x86_64.tar.gz
         #tar xzf cmake-*.tar.gz
         #mv cmake-*/CMake.app/Contents/* cmake
         #export PATH="$PWD/cmake/bin:$PATH"
@@ -192,10 +192,9 @@ before_install:
     elif [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
       PY_EXE=python$PYMVER
       if [[ -z "$DOCKER_BUILD" ]]; then
-        curl -L -O https://github.com/Kitware/CMake/releases/download/v3.13.1/cmake-3.13.1-Linux-x86_64.tar.gz
-        ls -l cmake-*
-        tar xzf cmake-*.tar.gz
-        mv cmake-*x86_64 cmake
+        curl -o cmake.tgz -L https://github.com/Kitware/CMake/releases/download/v3.13.4/cmake-3.13.4-Linux-x86_64.tar.gz
+        tar xzf cmake.tgz && rm cmake.tgz
+        ln -s cmake-*x86_64 cmake
         export PATH="$PWD/cmake/bin:$PATH"
       fi
     fi

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -1,4 +1,4 @@
-FROM ubuntu:16.04 as base
+FROM ubuntu:18.04 as base
 LABEL \
   author.name="Casper da Costa-Luis"          \
   author.email=imaging@caspersci.uk.to        \

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -76,7 +76,6 @@ COPY user_sirf-ubuntu.sh .
 COPY devel/.ccache/ /opt/ccache/
 RUN ccache -o cache_dir=/opt/ccache
 ARG BUILD_FLAGS="\
- -DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache\
  -DCMAKE_BUILD_TYPE=Release\
  -DBUILD_STIR_WITH_OPENMP=ON -DUSE_SYSTEM_ACE=ON\
  -DUSE_SYSTEM_Armadillo=ON -DUSE_SYSTEM_Boost=ON -DUSE_SYSTEM_FFTW3=ON\

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -76,10 +76,12 @@ COPY user_sirf-ubuntu.sh .
 COPY devel/.ccache/ /opt/ccache/
 RUN ccache -o cache_dir=/opt/ccache
 ARG BUILD_FLAGS="\
-  -DCMAKE_BUILD_TYPE=Release -DBUILD_STIR_WITH_OPENMP=ON -DUSE_SYSTEM_ACE=ON \
-  -DUSE_SYSTEM_Armadillo=ON -DUSE_SYSTEM_Boost=ON -DUSE_SYSTEM_FFTW3=ON \
-  -DUSE_SYSTEM_HDF5=ON -DUSE_ITK=ON -DBUILD_siemens_to_ismrmrd=ON \
-  -DUSE_SYSTEM_SWIG=ON"
+ -DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache\
+ -DCMAKE_BUILD_TYPE=Release\
+ -DBUILD_STIR_WITH_OPENMP=ON -DUSE_SYSTEM_ACE=ON\
+ -DUSE_SYSTEM_Armadillo=ON -DUSE_SYSTEM_Boost=ON -DUSE_SYSTEM_FFTW3=ON\
+ -DUSE_SYSTEM_HDF5=ON -DUSE_ITK=ON -DBUILD_siemens_to_ismrmrd=ON\
+ -DUSE_SYSTEM_SWIG=ON"
 ARG EXTRA_BUILD_FLAGS=""
 RUN bash user_sirf-ubuntu.sh
 RUN rm user_sirf-ubuntu.sh

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -76,9 +76,10 @@ COPY user_sirf-ubuntu.sh .
 COPY devel/.ccache/ /opt/ccache/
 RUN ccache -o cache_dir=/opt/ccache
 ARG BUILD_FLAGS="\
-  -DCMAKE_BUILD_TYPE=Release -DBUILD_STIR_WITH_OPENMP=ON \
+  -DCMAKE_BUILD_TYPE=Release -DBUILD_STIR_WITH_OPENMP=ON -DUSE_SYSTEM_ACE=ON \
   -DUSE_SYSTEM_Armadillo=ON -DUSE_SYSTEM_Boost=ON -DUSE_SYSTEM_FFTW3=ON \
-  -DUSE_SYSTEM_HDF5=ON -DBUILD_siemens_to_ismrmrd=ON -DUSE_SYSTEM_SWIG=ON"
+  -DUSE_SYSTEM_HDF5=ON -DUSE_ITK=ON -DBUILD_siemens_to_ismrmrd=ON \
+  -DUSE_SYSTEM_SWIG=ON"
 ARG EXTRA_BUILD_FLAGS=""
 RUN bash user_sirf-ubuntu.sh
 RUN rm user_sirf-ubuntu.sh

--- a/docker/build_essential-ubuntu.sh
+++ b/docker/build_essential-ubuntu.sh
@@ -14,9 +14,25 @@ apt-get install -yq --no-install-recommends \
   sudo
 apt-get clean
 
+pushd /opt
+
 # CMake
-mkdir /opt/cmake
-curl https://cmake.org/files/v3.7/cmake-3.7.2-Linux-x86_64.sh > cmake.sh
-echo y | bash cmake.sh --prefix=/opt/cmake --exclude-subdir
-export PATH="/opt/cmake/bin:$PATH"
-rm cmake.sh
+curl -o cmake.tgz -L https://github.com/Kitware/CMake/releases/download/v3.13.4/cmake-3.13.4-Linux-x86_64.tar.gz
+tar xzf cmake.tgz && rm cmake.tgz
+ln -s cmake-*x86_64 cmake
+export PATH="$PWD/cmake/bin:$PATH"
+
+# ccache
+mkdir -p bin
+pushd bin
+# ccache compiler override
+ln -s "$(which ccache)" g++
+ln -s "$(which ccache)" g++-6
+ln -s "$(which ccache)" g++-7
+ln -s "$(which ccache)" gcc
+ln -s "$(which ccache)" gcc-6
+ln -s "$(which ccache)" gcc-7
+export PATH="$PWD:$PATH"
+popd
+
+popd

--- a/docker/build_essential-ubuntu.sh
+++ b/docker/build_essential-ubuntu.sh
@@ -1,5 +1,6 @@
 #!/usr/bin/env bash
 set -ev
+INSTALL_DIR="${1:-/opt}"
 # Essential
 apt-get update -qq
 apt-get install -yq curl
@@ -14,7 +15,7 @@ apt-get install -yq --no-install-recommends \
   sudo
 apt-get clean
 
-pushd /opt
+pushd $INSTALL_DIR
 
 # CMake
 curl -o cmake.tgz -L https://github.com/Kitware/CMake/releases/download/v3.13.4/cmake-3.13.4-Linux-x86_64.tar.gz

--- a/docker/build_gadgetron-ubuntu.sh
+++ b/docker/build_gadgetron-ubuntu.sh
@@ -6,11 +6,4 @@ apt-get install -yq --no-install-recommends \
   h5utils              \
   liblapack-dev        \
   libace-dev
-# Not required (yet) by SIRF
-# libxml2-dev
-# libxslt-dev
-# python-h5py
-# python-libxml2
-# python-psutil
-# libplplot-dev
 apt-get clean

--- a/docker/build_gadgetron-ubuntu.sh
+++ b/docker/build_gadgetron-ubuntu.sh
@@ -1,9 +1,23 @@
 #!/usr/bin/env bash
 set -ev
 # Gadgetron
+# from https://github.com/gadgetron/gadgetron/blob/master/docker/base/ubuntu_1804/Dockerfile#L8
 apt-get update -qq
 apt-get install -yq --no-install-recommends \
-  h5utils              \
-  liblapack-dev        \
-  libace-dev
+  libhdf5-serial-dev     \
+  libboost-all-dev       \
+  libfftw3-dev           \
+  h5utils                \
+  jq                     \
+  hdf5-tools             \
+  libatlas-base-dev      \
+  libxml2-dev            \
+  libfreetype6-dev       \
+  libxslt-dev            \
+  libarmadillo-dev       \
+  libace-dev             \
+  liblapack-dev          \
+  liblapacke-dev         \
+  libplplot-dev          \
+  libdcmtk-dev
 apt-get clean

--- a/docker/build_system-ubuntu.sh
+++ b/docker/build_system-ubuntu.sh
@@ -3,25 +3,11 @@ set -ev
 # SIRF external dependencies
 apt-get update -qq
 apt-get install -yq --no-install-recommends \
-  hdf5-tools           \
   hdfview              \
-  libarmadillo-dev     \
   libboost-all-dev     \
-  libfftw3-dev         \
   libgtest-dev         \
-  libhdf5-serial-dev   \
-  libplplot-dev        \
-  libxml2-dev          \
-  libxslt-dev          \
   swig
 apt-get clean
 
 # Not required (yet) by SIRF
-# libxml2-dev
-# libxslt-dev
-# python-h5py
-# python-libxml2
-# python-psutil
-# libplplot-dev
 # libopenblas-dev
-# libatlas-base-dev

--- a/docker/build_system-ubuntu.sh
+++ b/docker/build_system-ubuntu.sh
@@ -7,4 +7,9 @@ apt-get install -yq --no-install-recommends \
   libfftw3-dev         \
   libhdf5-serial-dev   \
   swig
+
+# siemens_to_ismrmrd external dependencies
+apt-get install -yq --no-install-recommends \
+  libxml2-dev          \
+  libxslt-dev
 apt-get clean

--- a/docker/build_system-ubuntu.sh
+++ b/docker/build_system-ubuntu.sh
@@ -3,11 +3,8 @@ set -ev
 # SIRF external dependencies
 apt-get update -qq
 apt-get install -yq --no-install-recommends \
-  hdfview              \
   libboost-all-dev     \
-  libgtest-dev         \
+  libfftw3-dev         \
+  libhdf5-serial-dev   \
   swig
 apt-get clean
-
-# Not required (yet) by SIRF
-# libopenblas-dev

--- a/docker/build_system-ubuntu.sh
+++ b/docker/build_system-ubuntu.sh
@@ -15,3 +15,13 @@ apt-get install -yq --no-install-recommends \
   libxslt-dev          \
   swig
 apt-get clean
+
+# Not required (yet) by SIRF
+# libxml2-dev
+# libxslt-dev
+# python-h5py
+# python-libxml2
+# python-psutil
+# libplplot-dev
+# libopenblas-dev
+# libatlas-base-dev

--- a/docker/devel/test.sh
+++ b/docker/devel/test.sh
@@ -19,11 +19,12 @@ GCONFIG=./INSTALL/share/gadgetron/config/gadgetron.xml
 [ -f "$GCONFIG" ] || cp "$GCONFIG".example "$GCONFIG"
 ./INSTALL/bin/gadgetron >& gadgetron.log&
 
-# print for debugging
-[ "$DEBUG" != 0 ] && cat builds/SIRF/build/CMakeCache.txt
 ctest -VV
 ret=$?
+kill -n 15 $(pidof gadgetron)
+
 # print for debugging
+[ "$DEBUG" != 0 ] && cat builds/SIRF/build/CMakeCache.txt
 [ "$DEBUG" != 0 ] && cat builds/SIRF/build/Testing/Temporary/LastTest.log
 # may exceed 4MB travis log limit
 [ "$DEBUG" != 0 ] && cat gadgetron.log

--- a/docker/devel/test.sh
+++ b/docker/devel/test.sh
@@ -17,17 +17,17 @@ pushd $SIRF_PATH/../..
 # start gadgetron
 GCONFIG=./INSTALL/share/gadgetron/config/gadgetron.xml
 [ -f "$GCONFIG" ] || cp "$GCONFIG".example "$GCONFIG"
-./INSTALL/bin/gadgetron >& gadgetron.log&
+[ -f ./INSTALL/bin/gadgetron ] && ./INSTALL/bin/gadgetron >& gadgetron.log&
 
 ctest -VV
 ret=$?
-kill -n 15 $(pidof gadgetron)
+[ -n "$(pidof gadgetron)" ] && kill -n 15 $(pidof gadgetron)
 
 # print for debugging
 [ "$DEBUG" != 0 ] && cat builds/SIRF/build/CMakeCache.txt
 [ "$DEBUG" != 0 ] && cat builds/SIRF/build/Testing/Temporary/LastTest.log
 # may exceed 4MB travis log limit
-[ "$DEBUG" != 0 ] && cat gadgetron.log
+[ "$DEBUG" != 0 ] && [ -f gadgetron.log ] && cat gadgetron.log
 
 popd
 

--- a/docker/docker-compose.yml
+++ b/docker/docker-compose.yml
@@ -29,5 +29,5 @@ services:
    context: .
    target: core
    cache_from:
-    - ubuntu:16.04
+    - ubuntu:18.04
     - ccppetmr/sirf:core

--- a/docker/docker-compose.yml
+++ b/docker/docker-compose.yml
@@ -18,9 +18,12 @@ services:
     - ccppetmr/sirf:latest
   stdin_open: true
   tty: true
-  cap_add:
-   - NET_ADMIN
-  network_mode: "bridge"
+  # cap_add:
+  #  - NET_ADMIN
+  #TODO: uncomment for external gadgetron
+  # network_mode: "host"  # host only works on linux: https://docs.docker.com/network/host/
+  # depends_on:
+  #  - gadgetron
   volumes:
    - ./devel:/devel
  core:
@@ -34,9 +37,15 @@ services:
  gadgetron:
   container_name: gadgetron
   image: gadgetron/ubuntu_1804_no_cuda:latest
+  restart: unless-stopped
   environment:
-   GADGETRON_RELAY_HOST: sirf
+   GADGETRON_RELAY_HOST: 0.0.0.0
   ports:
-   - "9002:9002"  # supervisord web interface
+   - "9002:9002"  # gadgetron
    - "9080:9080"  # ReST API
    - "8002:8002"  # CloudBus relay
+   - "9888:8888"  # ? https://github.com/gadgetron/gadgetron/wiki/How-to-set-up-Gadgetron-cloud-in-the-LAN-environment
+   - "18002:18002"  # ReST API
+   - "9001:9001"  # supervisord web interface
+  volumes:
+   - ./devel/gadgetron_data:/tmp/gadgetron_data

--- a/docker/docker-compose.yml
+++ b/docker/docker-compose.yml
@@ -31,3 +31,12 @@ services:
    cache_from:
     - ubuntu:18.04
     - ccppetmr/sirf:core
+ gadgetron:
+  container_name: gadgetron
+  image: gadgetron/ubuntu_1804_no_cuda:latest
+  environment:
+   GADGETRON_RELAY_HOST: sirf
+  ports:
+   - "9002:9002"  # supervisord web interface
+   - "9080:9080"  # ReST API
+   - "8002:8002"  # CloudBus relay

--- a/docker/user_python-ubuntu.sh
+++ b/docker/user_python-ubuntu.sh
@@ -13,8 +13,8 @@ miniconda)
   rm miniconda.sh
   source "$INSTALL_DIR"/bin/activate
   conda config --add channels conda-forge
-  conda update -c conda-forge -y conda
-  conda update -c conda-forge -y --all
+  # https://github.com/conda/conda/issues/6030
+  #conda update -c conda-forge -y conda
   ;;
 *python*)
   # virtualenv

--- a/docker/user_python-ubuntu.sh
+++ b/docker/user_python-ubuntu.sh
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
-set -ev
 [ -f .bashrc ] && . .bashrc
+set -ev
 INSTALL_DIR="${1:-/opt/pyvenv}"
 PYTHON="${2:-miniconda}"
 

--- a/docker/user_sirf-ubuntu.sh
+++ b/docker/user_sirf-ubuntu.sh
@@ -1,9 +1,10 @@
 #!/usr/bin/env bash
 [ -f .bashrc ] && . .bashrc
 set -ev
+INSTALL_DIR="${1:-/opt}"
 # SIRF
-git clone https://github.com/CCPPETMR/SIRF-SuperBuild --recursive -b docker /opt/SIRF-SuperBuild
-pushd /opt/SIRF-SuperBuild
+git clone https://github.com/CCPPETMR/SIRF-SuperBuild --recursive -b docker $INSTALL_DIR/SIRF-SuperBuild
+pushd $INSTALL_DIR/SIRF-SuperBuild
 
 echo $BUILD_FLAGS $EXTRA_BUILD_FLAGS
 cmake $BUILD_FLAGS $EXTRA_BUILD_FLAGS .

--- a/docker/user_sirf-ubuntu.sh
+++ b/docker/user_sirf-ubuntu.sh
@@ -2,7 +2,7 @@
 [ -f .bashrc ] && . .bashrc
 set -ev
 # SIRF
-git clone https://github.com/CCPPETMR/SIRF-SuperBuild --recursive -b master /opt/SIRF-SuperBuild
+git clone https://github.com/CCPPETMR/SIRF-SuperBuild --recursive -b docker /opt/SIRF-SuperBuild
 pushd /opt/SIRF-SuperBuild
 echo $BUILD_FLAGS $EXTRA_BUILD_FLAGS
 cmake $BUILD_FLAGS $EXTRA_BUILD_FLAGS .

--- a/docker/user_sirf-ubuntu.sh
+++ b/docker/user_sirf-ubuntu.sh
@@ -4,11 +4,14 @@ set -ev
 # SIRF
 git clone https://github.com/CCPPETMR/SIRF-SuperBuild --recursive -b docker /opt/SIRF-SuperBuild
 pushd /opt/SIRF-SuperBuild
+
 echo $BUILD_FLAGS $EXTRA_BUILD_FLAGS
 cmake $BUILD_FLAGS $EXTRA_BUILD_FLAGS .
 make -j 2
 
 [ -f INSTALL/share/gadgetron/config/gadgetron.xml ] || \
+[ -f INSTALL/share/gadgetron/config/gadgetron.xml.example ] && \
   mv INSTALL/share/gadgetron/config/gadgetron.xml.example \
      INSTALL/share/gadgetron/config/gadgetron.xml
+
 popd

--- a/docker/user_sirf-ubuntu.sh
+++ b/docker/user_sirf-ubuntu.sh
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
-set -ev
 [ -f .bashrc ] && . .bashrc
+set -ev
 # SIRF
 git clone https://github.com/CCPPETMR/SIRF-SuperBuild --recursive -b master /opt/SIRF-SuperBuild
 pushd /opt/SIRF-SuperBuild

--- a/docker/user_sirf-ubuntu.sh
+++ b/docker/user_sirf-ubuntu.sh
@@ -4,6 +4,7 @@ set -ev
 # SIRF
 git clone https://github.com/CCPPETMR/SIRF-SuperBuild --recursive -b master /opt/SIRF-SuperBuild
 pushd /opt/SIRF-SuperBuild
+echo $BUILD_FLAGS $EXTRA_BUILD_FLAGS
 cmake $BUILD_FLAGS $EXTRA_BUILD_FLAGS .
 make -j 2
 

